beets/importer.py

```diff
@@ -1352,7 +1358,26 @@ def emitter(task):
         ])
         return pipeline.multiple(ipl.pull())
 
-    resolve_duplicates(session, task)
+    if type(task) != MergedImportTask:
+        resolve_duplicates(session, task)
+        if task.should_merge_duplicates:
+            duplicate_items = task.duplicate_items(session.lib)
+            for item in duplicate_items:
+                item.id = None
+                item.album_id = None
+
+            duplicate_paths = [item.path for item in duplicate_items]
+
+            merged_task = MergedImportTask(None, task.paths + duplicate_paths,
+                                           task.items + duplicate_items)
+
+            ipl = pipeline.Pipeline([
+                iter([merged_task]),
+                lookup_candidates(session),
+                user_query(session)
+            ])
```

Comment:
Similarly, maybe a helper function would be useful here to encapsulate the temporary pipeline as used here and in `user_query`.